HadesScheduler/HadesOperator/internal/controller/buildjob_containers.go

```
@@ -60,6 +61,66 @@ func (r *BuildJobReconciler) initializeContainerStatuses(ctx context.Context, bj
+// logsDrained reports whether all terminated containers of the BuildJob's pod have
+// had their logs published. podGone is true when the pod can no longer be found
+// (its logs are unrecoverable), so the caller should proceed to delete rather than
+// requeue forever.
+func (r *BuildJobReconciler) logsDrained(ctx context.Context, bj *buildv1.BuildJob) (drained bool, podGone bool, err error) {
+	pl := r.podLogReader(bj.Namespace, bj.Name)
+
+	podName, err := pl.ResolvePodName(ctx)
+	if err != nil {
+		// The pod cannot be resolved anymore; treat it as gone so we do not
+		// requeue forever waiting for logs that can no longer be read.
+		return false, true, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 99) but it returns nil

(nilerr)